eoxserver/services/ows/wps/processes/get_statistics.py

```diff
@@ -0,0 +1,142 @@
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies of this Software or works derived from this Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+# -----------------------------------------------------------------------------
+
+import numpy as np
+
+from eoxserver.core import Component, implements
+
+from eoxserver.contrib import gdal
+from eoxserver.services.ows.wps.interfaces import ProcessInterface
+from eoxserver.services.ows.wps.parameters import (
+    LiteralData, ComplexData, FormatJSON, CDObject
+)
+from eoxserver.services.ows.wps.exceptions import InvalidInputValueError
+
+from eoxserver.resources.coverages import models
+from eoxserver.backends.access import gdal_open
+from django.db.models import Q
+
+from django.contrib.gis.geos import Polygon
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class GetStatisticsProcess(Component):
+    """ GetStatistics defines a WPS process needed by the EOxC
+        DEM implementtation """
+
+    def __init__(self):
+        super(GetStatisticsProcess, self).__init__()
+
+    implements(ProcessInterface)
+
+    identifier = "GetStatistics"
+    title = "Get statistics for a coverage/s that intersects with the input bbox"
+    description = ("provides statistics of all the coverages whithin a provided bounding box. "
+                  " The process is used by the  by the EOxC DEM implementtation")
+    metadata = {}
+    profiles = ['EOxServer:GetStatistics']
+
+    inputs = {
+        "bbox": LiteralData(
+            "bbox",
+            title="bounding box that intersect with the products."
+        ),
+    }
+
+    outputs = {
+        "statistic": ComplexData(
+            "statistic",
+            title="output statistics",
+            abstract="coverage/s statistics in json format.",
+            formats=FormatJSON()
+        ),
+    }
+
+    @staticmethod
+    def execute(bbox, **kwarg):
+        """ The main execution function for the process.
+        """
+
+        values = list(map(float, bbox.split(",")))
+        if len(values) != 4:
+            raise ValueError("Invalid number of coordinates in 'bbox'.")
+
+        parsed_bbox = Polygon.from_bbox(values)
+
+        # get the dataset series intersecting with the requested bbox
+
+        coverages = models.Coverage.objects.filter(
+            Q(footprint__intersects=parsed_bbox)
+            | Q(footprint__isnull=True, parent_product__footprint__intersects=parsed_bbox))
+
```

Comment:
This is quite confusing: the only filter available is the footprint, but otherwise _all_ coverages stored will be selected. So for heterogeneous deployments this will most likely fail, when satellite imagery is mixed with DEM data.

To remedy this I'd suggest to either:
* Add a collection parameter to at least limit the search to that collection
* Add a coverage parameter to filter by coverage ID. In this case I'd make the bbox optional

Also the BBox parameter is only used for filtering, but not limiting the raster data eligible for statistics calculation. It would be a cool functionality to limit the area to the BBox.

Additionally it is a quite arbitrary drawback to limit the process to single-band files. The statistics could be calculated for multi-band files as well, just as easily.

The `NUMBER_OF_NODATA_PIXELS` calculation seems to be missing. The nodata value information could be retrieved from the `CoverageType` model of the coverage (in the `FieldType` model and associated `NilValue` structs).